dnf/module/repo_module_dict.py

```diff
@@ -0,0 +1,462 @@
+# Copyright (C) 2017  Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Library General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+import fnmatch
+import os
+from collections import OrderedDict
+
+import smartcols
+
+from dnf.conf.read import ModuleReader, ModuleDefaultsReader
+from dnf.exceptions import Error
+from dnf.module import module_errors, VERSION_LOCKED, NO_MODULE_ERR, STREAM_NOT_ENABLED_ERR, \
+    NO_ACTIVE_STREAM_ERR, INSTALLING_NEWER_VERSION, PROFILE_NOT_INSTALLED, NOTHING_TO_SHOW, \
+    NO_DEFAULT_STREAM_ERR
+from dnf.module.metadata_loader import ModuleMetadataLoader
+from dnf.module.repo_module import RepoModule
+from dnf.module.repo_module_version import RepoModuleVersion
+from dnf.module.subject import ModuleSubject
+from dnf.util import first_not_none, logger, ensure_dir
+
+
+class PreferredModuleVersion(object):
+
+    def __init__(self):
+        self.moduleversion_nsvcap_pkgspec = []
+        self.preferred_version = -1
+        self.reason = None
+
+
+class RepoModuleDict(OrderedDict):
```

Comment:
RepoModuleDict() and RepoModule() is combination of pase.API and Cli, because some of action requires confirmation by user or assume yes or no.